File: src/main/java/com/github/bjoernpetersen/jmusicbot/playback/DummyPlayback.java
package com.github.bjoernpetersen.jmusicbot.playback;

/**
 * A dummy implementation of Playback, only used by {@link Player}.
 */
final class DummyPlayback implements Playback {

  private static final DummyPlayback INSTANCE = new DummyPlayback();

  public static DummyPlayback getInstance() {
    return INSTANCE;
  }

  @Override
  public void play() {
  }

  @Override
  public void pause() {
  }

  @Override
  public void waitForFinish() throws InterruptedException {
  }

  @Override
  public void close() throws Exception {
  }
}

File: src/main/java/com/github/bjoernpetersen/jmusicbot/playback/DummyPlayback.kt
package com.github.bjoernpetersen.jmusicbot.playback


object DummyPlayback : Playback {
    override fun play() {}

    override fun pause() {}

    @Throws(InterruptedException::class)
    override fun waitForFinish() {
        Thread.sleep(2000)
    }

    override fun close() {}

}